controllers/statusreport/statusreport_adapter.go

```diff
@@ -109,6 +110,17 @@ func (a *Adapter) EnsureSnapshotFinishedAllTests() (controller.OperationResult,
 		return controller.RequeueWithError(err)
 	}
 
+	// If for the snapshot there is an IntegrationTestScenario that is not triggered, it will add run labebl to snapshot
+	integrationTestScenarioNotTriggered := a.determineIfAllIntegrationTestsNotTriggered(integrationTestScenarios, testStatuses)
+	if integrationTestScenarioNotTriggered != "" {
+		a.logger.Info("Detected an integrationTestScenario was not triggered, applying snapshot reconcilation",
+			"integrationTestScenario.Name", integrationTestScenarioNotTriggered)
+		if err = gitops.AddIntegrationTestRerunLabel(a.client, a.context, a.snapshot, integrationTestScenarioNotTriggered); err != nil {
+			return controller.RequeueWithError(err)
+		}
+
+	}
```

Comment:
I'm thinking that  moving this logic to be inside the condition when not all integration tests finished ([here](https://github.com/redhat-appstudio/integration-service/blob/dd5a5e7abc4e575de00d5b568b95a8f74a13a50d/controllers/statusreport/statusreport_adapter.go#L120) currently) would make sense since there's no need to check this if we determine that everything passed.

Reply:
I think it wouldn't make sense to add it there as it would hide the purpose of the `determineIfAllIntegrationTestsFinishedAndPassed` function (by making it do more than expected) and would it make harder to maintain in the future. 

That's why I proposed adding it [here](https://github.com/redhat-appstudio/integration-service/blob/dd5a5e7abc4e575de00d5b568b95a8f74a13a50d/controllers/statusreport/statusreport_adapter.go#L120) because at that point we know that not all scenarios finished and can then run the new logic to find missing pipelineRuns in a place that's more visible.

Reply:
agree, 
moved it to [here](https://github.com/redhat-appstudio/integration-service/blob/dd5a5e7abc4e575de00d5b568b95a8f74a13a50d/controllers/statusreport/statusreport_adapter.go#L120)

Reply:
Thanks!